### DISCUSSION TITLE
Alpha/Beta -> API stability?

### DISCUSSION THREAD

**Discussion by waisaacs** [raspberrypi/picamera2#278] [General]
First, this is a thank you note.  All the hard work is noticeable and appreciated.  I'm looking forward to trying out this most recent set of enhancements.

So, I noticed the changelog calls the 0.3.x release Beta, but the readme still has the Alpha release warning language regarding API stability.  Just wondering if you've crossed an internal threshold where you feel like interfaces have settled.  I know you've commented about work still to be done around managing timing of camera controls changes taking effect, so it feels like substantial stuff is still coming, but maybe that's all under the hood at this point.  Low priority question out of curiosity, not like I'm filing a documentation bug!

**Comment by davidplowman** [comment]
Hi, thanks for pointing out about the readme, it obviously escaped my attention and I'll get that updated at some point.

Software developers are notorious for the fact that they never want to stop fiddling, but I've called this release "beta" rather than "alpha" because that really does need to stop, or at least slow down significantly. I can't promise that there won't be any further changes (libcamera itself doesn't even have a formally stable API), and bug fixes are inevitable of course, but non-backwards-compatible API changes will be very heavily frowned upon.

You correctly raise the point about the delays for control changes to take effect. That's got a bit held up with folks being away on summer holidays, but progress has been made. I think some API changes are inevitable here, but I'm hopeful that the route chosen will allow them not to break existing behaviour. Watch this space...!
